src/freenet/clients/http/WelcomeToadlet.java

```diff
@@ -414,33 +414,37 @@ public void handleMethodGET(URI uri, HTTPRequest request, ToadletContext ctx) th
 			addCategoryToList(BookmarkManager.DEFAULT_CATEGORY, bookmarksList, (!container.enableActivelinks()) || (useragent != null && useragent.contains("khtml") && !useragent.contains("chrome")), ctx);
 		}
 
-		// Search Box
-        // FIXME search box is BELOW bookmarks for now, until we get search fixed properly.
-		HTMLNode searchBox = contentNode.addChild("div", "class", "infobox infobox-normal");
-		searchBox.addAttribute("id", "search-freenet");
-        searchBox.addChild("div", "class", "infobox-header").addChild("span", "class", "search-title-label", NodeL10n.getBase().getString("WelcomeToadlet.searchBoxLabel"));
-		HTMLNode searchBoxContent = searchBox.addChild("div", "class", "infobox-content");
-		// Search form
-		if(node.pluginManager != null &&
-				node.pluginManager.isPluginLoaded("plugins.Library.Main")) {
-        	// FIXME: Remove this once we have a non-broken index.
-        	searchBoxContent.addChild("span", "class", "search-warning-text", l10n("searchBoxWarningSlow"));
-			HTMLNode searchForm = container.addFormChild(searchBoxContent, "/library/", "searchform");
-        	searchForm.addChild("input", new String[] { "type", "size", "name" }, new String[] { "text", "80", "search" });
-        	searchForm.addChild("input", new String[] { "type", "name", "value" }, new String[] { "submit", "find", l10n("searchFreenet") });
-        	// Search must be in a new window so that the user is able to browse the bookmarks.
-        	searchForm.addAttribute("target", "_blank");
-        } else if(node.pluginManager == null || 
-        		node.pluginManager.isPluginLoadedOrLoadingOrWantLoad("Library")) {
-			// Warn that search plugin is not loaded.
-			HTMLNode textSpan = searchBoxContent.addChild("span", "class", "search-not-availible-warning");
-			NodeL10n.getBase().addL10nSubstitution(textSpan, "WelcomeToadlet.searchPluginLoading", new String[] { "link" }, new HTMLNode[] { HTMLNode.link("/plugins/") });
-        } else {
-			// Warn that search plugin is not loaded.
-			HTMLNode textSpan = searchBoxContent.addChild("span", "class", "search-not-availible-warning");
-			NodeL10n.getBase().addL10nSubstitution(textSpan, "WelcomeToadlet.searchPluginNotLoaded", new String[] { "link" }, new HTMLNode[] { HTMLNode.link("/plugins/") });
-		}
-		
+        // Search Box: Only show it if Library is loaded.
+        if(node.pluginManager != null &&
+           node.pluginManager.isPluginLoaded("plugins.Library.Main")) {
+            // search box is BELOW bookmarks for now, until we get search fixed properly.
+            HTMLNode searchBox = contentNode.addChild("div", "class", "infobox infobox-normal");
```

Comment:
Maybe addSearchBoxForLibrary(), since it adds the box to the contentNode?

Do I understand it correctly that  you mean replacing the whole affected code by something like

    if (showSearchBox()) {
        addSearchBox();
    }